src/TypeInference/Uniqueness.ml

```diff
@@ -48,7 +48,7 @@ let check_inst_uniqueness insts =
   let name_of (inst : S.inst) =
     match inst.data with
     | (NLabel, _) -> "label"
-    | ((NVar n | NImplicit n), _) -> n
+    | ((NVar n | NImplicit n | NMethod n), _) -> n
```

Comment:
I'm not sure if this is right. Methods are identified by name and owner, so one can imagine the case when two methods of the same name are explicitly instantiated, but they can be distinguished by the owner type. On the other hand, I do not see any practical scenario, where user might want to explicitly instantiate a method, so what is implemented is good enough.